### PR TITLE
Use cwd and create results directory if it doesn't exist.

### PR DESCRIPTION
**Pull request by jbenjoseph**
Addresses #63

### CODE REVIEW

**Comment by jspeed-meyers** [comment]
@jbenjoseph, thank you for working on this. One question: Am I correct that this functionality places the results directory in your current path? For instance, I pip install the package and call it while the desktop is my working directory. I can then expect the results directory to be on my desktop?

**Comment by jbenjoseph** [comment]
That is the expected behavior.

**Comment by jspeed-meyers** [comment]
@jbenjoseph, okay, good. Should we make the mapping functionality also save in the same way?

https://github.com/IQTLabs/GitGeo/blob/24d901612cd3825cf6971d162d3dab1078b12867/gitgeo/mapping.py#L97

We could open a new issue or combine it here. Up to you.

**Comment by jspeed-meyers** [comment]
@jbenjoseph, any idea why the tests are failing?

**Comment by jbenjoseph** [comment]
This appears to be due to one of the functions that take paths as input expected a ```str``` as opposed as to a ```Path-like object```. This also has been fixed.

**Comment by jspeed-meyers** [comment]
@jbenjoseph, one question: do you think this functionality is worth turning into a separate function? Or is that overkill? I defer to your judgment.

**Comment by jspeed-meyers** [comment]
Okay, I did just explore this current branch, looking for any bugs and ran into one, though I don't think this bug is related to this new functionality.

Here's the bug:

```gitgeo --repo www.github.com/psf/requests --map```

leads to:

```
Traceback (most recent call last):
  File "/Users/jmeyers/opt/anaconda3/bin/gitgeo", line 33, in <module>
    sys.exit(load_entry_point('gitgeo==1.0.1', 'console_scripts', 'gitgeo')())
  File "/Users/jmeyers/opt/anaconda3/lib/python3.7/site-packages/gitgeo-1.0.1-py3.7.egg/gitgeo/main.py", line 119, in main
    make_map(repo=args.repo, num=args.num)
  File "/Users/jmeyers/opt/anaconda3/lib/python3.7/site-packages/gitgeo-1.0.1-py3.7.egg/gitgeo/mapping.py", line 47, in make_map
    world_json = add_contributor_count_to_json(df)
  File "/Users/jmeyers/opt/anaconda3/lib/python3.7/site-packages/gitgeo-1.0.1-py3.7.egg/gitgeo/mapping.py", line 117, in add_contributor_count_to_json
    with open("world.json") as original_json_file:
FileNotFoundError: [Errno 2] No such file or directory: 'world.json'
```

I suspect the world.json file (A) ought to be placed into the gitgeo directory for conceptual coherence and (B) then referenced appropriately.

Question to you @jbenjoseph: Do you want me to open a separate issue for this? Or do you want to combine the commit that squashes this bug here?

(Also, tangent: I'm really glad that you are helping make GitGeo a library (not just an application) and also installable from PyPI. Great ideas!)

File: gitgeo/custom_csv.py [hunk 1]
```diff
@@ -17,7 +17,9 @@ def create_csv(results_type, timestamp):
     Returns:
         None
     """
-    filename = os.path.join("results", results_type + "_" + timestamp + ".csv")
+    filename = Path.cwd() / "results"
+    filename.mkdir(exist_ok=True)
+    filename /= results_type + "_" + timestamp + ".csv"
```
Comment (jspeed-meyers) [inline gitgeo/custom_csv.py:22]: @jbenjoseph, one question here: What is this ```\=``` in the context of a string operation? Google failed me here.

Comment (jbenjoseph) [reply gitgeo/custom_csv.py:22]: This is overloaded by the ```pathlib``` for creating paths in a cross-platform manner. Using ```/=``` is equivalent to  ```filename = filename / results_type + "_" + timestamp + ".csv"```

**Comment by jbenjoseph** [comment]
@jspeed-meyers New bug addressed. Most of these bugs are related the code expecting files being in certain places which no longer became true when the code as converted into a package. I've defined any files I found like this as package-relative as opposed to working-directory relative, which means the command should work from any directory.

**Comment by jspeed-meyers** [comment]
@jbenjoseph, this is awesome. Nice work!

**Pull request merged**